### PR TITLE
clean up elbo gradient subsample test

### PR DESCRIPTION
**Pull request by martinjankowiak**
next: add parallelized version and other variants

### CODE REVIEW

**Comment by martinjankowiak** [comment]
should be ready to merge (no integration test changes)

**Comment by jpchen** [comment]
is this good for merge? it looks fine to me, though i think with the new pytorch master all `variable`s should be `tensor`, not sure if theres a separate effort to do a codebase wide sweep

**Comment by fritzo** [comment]
> @jpchen i think with the new pytorch master ...

@neerajprad has carefully pinned our CI builds to commit 8327982 which supports both `variable` and `torch.tensor`. This will allow us to gradually refactor the Pyro codebase over the next few days. See discussion in #857 for @neerajprad's refactoring plan.
